### PR TITLE
Add touch functionality to snap.svg.zpd

### PR DESCRIPTION
**Pull request by deadosaurus**
- Add single touch panning and dragging.
- Add multi-touch zooming for 2 touches only. This can be changed to work for more touches by changing this line 'event.touches.length == 2'. Be aware that the zooming calculation will only work based off the first 2 touches.

### CODE REVIEW

**Comment by huei90** [comment]
This is really cool!! @deadosaurus awesome.

But do you mind make the commit one by one what you have done? 

I'm not able to see your code in current diff.

**Comment by deadosaurus** [comment]
@huei90 sure I will do so. I don't use Github, so still learning :)

**Comment by huei90** [comment]
Thank you @deadosaurus take time.
feel free to use git to commit it 😄

**Comment by deadosaurus** [comment]
@huei90 I think that has worked, take a look when you get a chance.

**Comment by huei90** [comment]
Do you mind creating that the `touch` as the option to allow on/off touchable feature

 given below example

```js
paper.zpd({
   touch: false // default given true
});
```

@tikiatua Have a look of this PR.

🍻

**Comment by deadosaurus** [comment]
@huei90 sure, I can do that. Will commit changes later today.

File: snap.svg.zpd.js [hunk 1]
```diff
@@ -403,50 +464,77 @@ SVGElement.prototype.getTransformToElement = SVGElement.prototype.getTransformTo
...
+                    var distance = _getMultiTouchDistance(event);
 
-                if (typeof(stateTf) == 'undefined') {
-                    zpdElement.data.stateTf = g.getCTM().inverse();
-                }
+                    if (zpdElement.data.prevZoomDistance != null) {
+                        var delta = 0;
+
+                        if (zpdElement.data.prevZoomDistance > distance) delta = -0.1111111111111111;
+                        else delta = 0.1111111111111111;
```
Comment (huei90) [inline snap.svg.zpd.js]: why of this `0.1111111111111111` magic number?

Comment (deadosaurus) [inline snap.svg.zpd.js]: Yes the magic number, works well on iPad, I don't have many other touch devices to test it on.

File: snap.svg.zpd.js [hunk 2]
```diff
@@ -403,50 +464,77 @@ SVGElement.prototype.getTransformToElement = SVGElement.prototype.getTransformTo
...
+                "mouseWheel": handleMouseWheel,
+                "touchMove": handleTouchMove
             };
         };
 
 
         /**
          * Register handlers
-         * desktop and mobile (?)
+         * desktop and mobile
```
Comment (huei90) [inline snap.svg.zpd.js]: 😄 Thanks for this tiny typo modification.

**Comment by deadosaurus** [comment]
Delta value of 0.15 works well on both iPad and Samsung S6.

File: snap.svg.zpd.js [hunk 3]
```diff
@@ -403,50 +464,77 @@ SVGElement.prototype.getTransformToElement = SVGElement.prototype.getTransformTo
...
+                    var distance = _getMultiTouchDistance(event);
 
-                // Compute new scale matrix in current mouse position
-                var k = zpdElement.data.root.createSVGMatrix().translate(p.x, p.y).scale(z).translate(-p.x, -p.y);
+                    if (zpdElement.data.prevZoomDistance != null) {
+                        var delta = 0;
 
-                _setCTM(g, g.getCTM().multiply(k), zpdElement.options.zoomThreshold);
+                        if (zpdElement.data.prevZoomDistance > distance) delta = -0.15;
+                        else delta = 0.15;
```
Comment (huei90) [inline snap.svg.zpd.js]: Do you mind add a comment here to explain why this magic numbers 0.15 or 0.111111.
Useful for future code reading 😃

**Comment by huei90** [comment]
It will be great if you can add the following in README.md
https://github.com/huei90/snap.svg.zpd#options

---

#### drag

```
true or false: enable or disable touchable (default true)
```

**Comment by huei90** [comment]
So far looks good, now waiting for the contributor @tikiatua .

**Comment by tikiatua** [comment]
Hi there,

many thanks for adding this. I will take a look and test it on different devices next week.

**Comment by deadosaurus** [comment]
@tikiatua & @huei90 pleasure, but thank you for the code to begin with!

**Comment by huei90** [comment]
@tikiatua If everything good, I'm going to merge it.

**Comment by deadosaurus** [comment]
We are using this in production so if there are any bugs, I will fix them.

**Comment by huei90** [comment]
ok then merge! 🍻

**Pull request merged**